test/middleware-test.ts

```diff
@@ -0,0 +1,710 @@
+// /test/middleware-test.ts
+// Tests the rate limiting middleware
+
+import { jest } from '@jest/globals'
+import Express from 'express'
+import Sinon, { SinonFakeTimers } from 'sinon'
+import request from 'supertest'
+
+import rateLimit, {
+	Store,
+	Options,
+	IncrementResponse,
+} from '../dist/esm/index.js'
+
+import { createServer } from './helpers/create-server.js'
+
+const sandbox = Sinon.createSandbox()
+
+describe('middleware test', () => {
+	let clock: SinonFakeTimers
+	beforeEach(() => {
+		clock = Sinon.useFakeTimers()
+	})
+	afterEach(() => {
+		clock.restore()
+		sandbox.restore()
```

Comment:
since you're doing `jest.spyOn` instead of `sandbox.stub` in some places below, we need to add a `jest.restoreAllMocks()` here.
```suggestion
		sandbox.restore()
		jest.restoreAllMocks()
```

Reply:
Should I replace the `spy.mockRestore` call in the `finally` block with this instead?

Reply:
Yeah, I'd kind of prefer that.